src/snowflake/snowpark/mock/snowflake_data_type.py

```diff
@@ -435,25 +446,28 @@ def __round__(self, n=None):
 
     def __rpow__(self, other):
         result = super().__rpow__(other)
-        result.sf_type = ColumnType(DoubleType(), self.sf_type.nullable)
+        result.sf_type = ColumnType(FloatType(), True)
         return result
 
     def __rtruediv__(self, other):
-        result = super().__rtruediv__(other)
-        result.sf_type = calculate_type(other.sf_type, self.sf_type, op="/")
-        return result
+        return other.__truediv__(self)
 
     def __truediv__(self, other):
         result = super().__truediv__(other)
-        result.sf_type = calculate_type(self.sf_type, other.sf_type, op="/")
+        sf_type = calculate_type(self.sf_type, other.sf_type, op="/")
+        if isinstance(sf_type.datatype, DecimalType):
+            result = result.astype("double").round(sf_type.datatype.scale)
+        elif isinstance(sf_type.datatype, (FloatType, DoubleType)):
+            result = result.astype("double").round(16)
```

Comment:
We observe that Snowflake returns 16 digits after the floating point. cc: @sfc-gh-yixie